src/test/java/com/session/service/ClientTest.java

```diff
@@ -67,14 +70,42 @@ public void createSession_emptyEmailAddress_shouldReturnError() {
     }
 
     @Test
-    public void getSessionById_shouldReturnExpectedSession() throws Exception {
-        SessionCreated sessionCreated = client.createNewSession(EMAIL);
-        assertThat(sessionCreated, is(notNullValue()));
-        assertThat(sessionCreated.getId(), not(isEmptyString()));
+    public void getSessionById_shouldReturnExpectedSession() throws IOException {
+        Mockito.when(http.get(
+                eq(HOST),
+                eq("/sessions/" + SESSION_ID),
+                ArgumentMatchers.<ResponseHandler<Session>>any()
+        )).thenReturn(zebedeeSession);
 
-        Session session = client.getSessionByID(sessionCreated.getId());
+        Session session = client.getSessionByID(SESSION_ID);
```

Comment:
[Personal opinion not request for change] I personally like to have each section of the test grouped together:
- Set up
- Execution
- Assertions 
```
        // Set up
        Mockito.when(http.post(
                eq(HOST),
                eq("/sessions"),
                any(Object.class),
                ArgumentMatchers.<ResponseHandler<SessionCreated>>any(),
                any(String.class)
        )).thenReturn(new SessionCreated(RETURNED_URI, SESSION_ID));

        // Execution
        SessionCreated sessionCreated = client.createNewSession(EMAIL);

       // Assertions
        assertThat(sessionCreated, is(notNullValue()));
        assertThat(sessionCreated.getId(), is(SESSION_ID));

```
This is a fairly trivial test so the value isn't immediately obvious. When tests become bigger and the set up/assertions become more complex I think it helps to keep the test readable.

All just my opinion though.

Functionally this test is 👌